src/Symfony/Component/Security/Http/Attribute/IsGranted.php

```diff
@@ -24,15 +28,15 @@
 final class IsGranted
 {
     /**
-     * @param string|Expression            $attribute     The attribute that will be checked against a given authentication token and optional subject
-     * @param array|string|Expression|null $subject       An optional subject - e.g. the current object being voted on
-     * @param string|null                  $message       A custom message when access is not granted
-     * @param int|null                     $statusCode    If set, will throw HttpKernel's HttpException with the given $statusCode; if null, Security\Core's AccessDeniedException will be used
-     * @param int|null                     $exceptionCode If set, will add the exception code to thrown exception
+     * @param string|Expression|(\Closure(TokenInterface $token, mixed $subject, AccessDecisionManagerInterface $accessDecisionManager, AuthenticationTrustResolverInterface $trustResolver): bool) $attribute     The attribute that will be checked against a given authentication token and optional subject, or a callable that will be called to determine access
```

Comment:
if we go with the variadic call, we might document this as this:
```suggestion
     * @param string|Expression|(\Closure(array{'token': TokenInterface, 'subject': mixed, etc} ...array $vars): bool) $attribute     The attribute that will be checked against a given authentication token and optional subject
```

Reply:
no. there is no way to describe a signature using a variadic argument to emulate unordered named parameters (which is quite a hacky way to write a signature)

Reply:
php erroring will make it clean enough + it's already mentionned in the phpdoc